sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/component.py

```diff
@@ -110,6 +110,28 @@ def _serialize(self, obj, *, many: bool = False):
                 ret[attr_name] = self.get_attribute(obj, attr_name, None)
         return ret
 
+    @pre_load()
+    def convert_input_value_to_str(self, data: dict, **kwargs) -> dict:
+        """
+        Convert the non-str value in input to str.
+
+        When load the v1.5 component yaml, true/false will be converted to bool type and yyyy-mm-dd will be
+        converted to date type. In order to be consistent with before, it needs to be converted to str type.
+        """
+        def convert_to_str(value):
+            if isinstance(value, bool):
+                return str(value).lower()
+            return str(value)
+
+        if "inputs" in data and isinstance(data["inputs"], dict):
+            for input_port in data["inputs"].values():
+                if input_port["type"] in ["String", "Enum"]:
```

Comment:
could user write type as something like "string"?

Reply:
Fix it